src/libraries/System.Private.Xml/src/System/Xml/Schema/XmlSchema.cs

```diff
@@ -117,6 +118,10 @@ public void Write(XmlWriter writer)
             Write(writer, null);
         }
 
+        [DynamicDependency(TrimmerConstants.PublicMembers, typeof(XmlSchema))]
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+            Justification = "Supressing warning since the right members of XmlSchema will be preserved by the " +
+            "DynamicDependency attribute.")]
```

Comment:
This is not enough unfortunately. Serialization is recursive - the dynamic dependency you added will only work on `XmlSchema` itself, not any of the other types the serializer will get to by recursively serialization data on the schema. @sbomer and/or @LakshanF are working on adding some support for `XmlSerializer` based on custom steps. I *think* it could make this work if fully implemented (the `XmlSchema` has `XmlRootAttribute` on it, so it's discoverable by the custom step). I don't know how far we've got though.

Reply:
All that said I would probably look at prioritizing this. I don't think it's very common to write `xsd` files in general - and in trimmed app on top...

Reply:
I see, I'm not very familiar with XmlSchema but do you have an example of a type of Schema that would not work with just this DynamicDependency so that I can add a trimming test for it and to figure out what else needs to be preserved? I made sure that the trimming test I added in the last commit was failing without the DynamicDependency attribute and it is passing now after adding that annotation.

Reply:
I'd have to play with it - but high level:
* The XmlSchema creates an OM representation of the schema in memory
* The OM is basically a graph of XmlSchemaObject and derived classes (plus probably some other helper classes and so on)
* The serialization will walk that OM and write it out into XML - in great detail (after all it should be an exact serialization - meaning you should be able to roundtrip the OM through the file).
* That means the serialization will need pretty the entire OM preserved

Basically I would have to go and find a hidden property somewhere for which nobody calls a getter - create the object it belongs to and pass it to the serialization -> boom.
I guess what I'm trying to say - this can be pretty costly to get right - we would need to somehow figure out (and ideally prove) the full graph the serializer can walk in this case.

Reply:
I'd be fine with adding the test, `DynamicDependency`, and `RequiresUnreferencedCode` for now. Users will still get a warning if they are calling this to let them know it may not work. We can always remove the `RequiresUnreferencedCode` in the future once we are 100% certain everything will work correctly.

Reply:
I like @eerhardt suggestion.